### DISCUSSION TITLE
Sync-able Off-line Data Platform

### DISCUSSION THREAD

**Discussion by valschmidt** [foxglove/community#600] [Q&A]
We deploy our robots from ships at sea, and often have intermittent or poor connections to shore. But our deployed operators still need the ability to view logs for troubleshooting and maintenance activities. We use Foxglove Studio for this to great effect, but it suffers from an inability to view multiple log files at once. As subscribers to the Data Platform, we really like its ability to work across data files. 

So what we would like is the ability to install a local version of the Foxglove Data Platform on a local computer. Here's our envisioned workflow:

* Robot is deployed.
* Possibly during deployment, but more likely after robot is recovered, our automated script pushes our operational log files to the Local Foxglove Data Platform. (LFDP)
* The LFDP parses that data, just as is done in the cloud based service, and provides a similar user interface to it. 
* Users can view and inspect log data served up by the LFDP locally just as the do with the cloud based service.
* When ship to shore bandwidth allows (possibly on return to port) the contents of the LFDP is pushed to our cloud Foxglove Data Platform account, along with any Events that were created on the LFDP.

**Comment by defunctzombie** [comment]
@valschmidt We actually have a feature with data platform that addresses some of the items above. We call this feature "edge sites" (https://foxglove.dev/docs/data-platform/edge-sites/introduction). Today edge sites address the last bullet point and allow you to request specific data sync to primary cloud sites when internet connectivity permits it. Unfortunately they do not support local streaming or browsing in an off-line manner like you are expecting from the console interface. Please reach out to us via our contact us form: https://foxglove.dev/contact and we can chat more about your use case and product capabilities.

**Reply by valschmidt** [reply]
Thanks!
